### ISSUE TITLE
TS compiler accept incorrect code

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/TypeScript-TmLanguage#426]
**Version**: 2.2.1

**Code**

```ts
interface DocumentFilter {
	language?: string;
	scheme?: string;
	pattern?: string;
}

type DocumentSelector = DocumentFilter[];

let selector: DocumentFilter | string[];

selector = [10];
selector = ['bat', { language: 'ts', scheme: 'file'}];
```

Both assignments to `selector` should produce an error.

**Comment by dbaeumer** [comment]
Ignore it. The problem is that 10 matches DocumentFilter since it accepts a filter with no properties.
